joomla/joomla-cms
# [com_joomlaupdate] Check for minimum supported database type and version

Pull Request for Issue #12351
### Summary of Changes

This implements a minimum database feature into com_joomlaupdate so we can check for minimum supported database version per type
### Testing Instructions

First Please review the code.
#### Test the Core
- Install 3.7.x (https://github.com/joomla/joomla-cms/archive/3.7.x.zip)
- set this as custom update server `http://www.jah-tz.de/downloads/core/nightlies/next_minor_list.xml`
- confirm that com_joomlaupdate still points to 3.7.0-dev2
- apply this patch
- confirm that com_joomlaupdate still points to 3.7.0-dev2
- Do NOT install this update
- change the custom update server to `http://www.jah-tz.de/downloads/core/nightlies/next_minor_list2.xml`
- confirm that you get the `We can't find a download URL ....` message in com_joomlaupdate
- change the custom update server to `http://www.jah-tz.de/downloads/core/nightlies/next_minor_list3.xml`
- confirm that com_joomlaupdate still points to 3.7.0-dev2 (if you meat the requirements of mysql 5.5.3 else you should get the warning)
#### Test Extensions
- install 3.7.x (https://github.com/joomla/joomla-cms/archive/3.7.x.zip)
- install an older component of choice e.g. JCE
- confirm that the update show and run as expected
- install this custom component (http://www.jah-tz.de/downloads/download.php?file=plg_dbminimum.zip&folder=core/extensions/plg_dbminimum/)
- confirm that it adds this update server `http://www.jah-tz.de/downloads/core/extensions/plg_dbminimum/plg_dbminimum.xml` (see: Backend -> Extensions -> Manage -> Update Sites)
- Try to update to the next version (0.0.2)
- confirm that you got a error message
### Documentation Changes Required

Document the new tag if implemented as this can also be used by 3party extensions
